Enable primitives `serde` feature in archiving

vedhavyas opened the pull request:
This PR enables serde feature in archiving that was missing and therefore causing the build to fail. More here - https://github.com/subspace/subspace/actions/runs/4991087910/jobs/8937321575#step:3:2292


I'll create a companion PR to main once this PR lands


### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

nazar-pc reviewed (changes requested):
Nope, this is not the way to go. `serde` here is a feature. It can only be enabled in `std`, not in `no_std`. But both `std` and `no_std` compile just fine without `serde` feature enabled. In case `dep:serde` is enabled (the only way to do that is via `serde` feature), then `serde?/std` will get enabled too as the result.

PR description talks about anything except the reason why this change is necessary, could you elaborate why this change is necessary in the first place?

vedhavyas committed:
enable subspace-primitives serde in archiving

Turns out `primitives serde` was not included in the std and hence some types had missing serde derivations.

nazar-pc reviewed (approved):
This looks good, thanks!
